packages/caravan-health/src/privacy.ts

```diff
@@ -0,0 +1,204 @@
+import {BlockchainClient} from '@caravan/clients'
+/*
+The methodology for calculating a privacy score (p_score) for Bitcoin transactions based 
+on the number of inputs and outputs is the primary point to define wallet health for privacy. 
+The score is further influenced by several factors such as address reuse, 
+address types and UTXO set fingerprints etc.
+
+More on the algorithms for scoring privacy: <TODO : Add Link to Blog Post>
+*/
+
+// A normalizing quantity that increases the score by a certain factor in cases of self-payment.
+// More about deniability : https://www.truthcoin.info/blog/deniability/
+const DENIABILITY_FACTOR = 1.5;
+
+/*
+The p_score is calculated by evaluating the likelihood of self-payments, the involvement of 
+change outputs and the type of transaction based on number of inputs and outputs.
+
+We have 5 categories of transaction type
+- Sweep Spend
+- Simple Spend
+- UTXO Fragmentation
```

Comment:
So what is UTXO fragmentation?

Reply:
A transaction type where a larger UTXO is broken into small chunks. By definition, fragmentations as a word fits for such behaviour.

Reply:
Right, so in my mind I was thinking since you're breaking down a utxo into multiple smaller pieces then the assumption it's not changing ownership just changing size basically. Alternatively, you could define a UTXO fragmentation as any transaction with more than 2 outputs? (maybe put that in parentheses if that's the case)
```suggestion
- UTXO Fragmentation (any transaction with more than the standard 2 outputs)
```